src/structures/TextChannel.js

```diff
@@ -35,7 +35,7 @@ class TextChannel extends GuildChannel {
      * @type {boolean}
      * @readonly
      */
-    this.nsfw = Boolean(data.nsfw);
+    this.nsfw = data.nsfw || /^nsfw(-|$)/.test(this.name);
```

Comment:
The regex should be a variable instead of creating it every time the function is called